etc/systemd/system/zfs-import-cache.service.in

```diff
@@ -12,8 +12,9 @@ ConditionPathExists=@sysconfdir@/zfs/zpool.cache
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStartPre=/sbin/modprobe zfs
+ExecStartPre=@sbindir@/modprobe zfs
```

Comment:
We don't provide/package the `modprobe` binary, AFAIK `@sbindir@` is only for "our" executables.

Reply:
sbindir contains a bunch of files in dracut.  also at least on my system after boot /sbin/modprobe is valid

```
find /var/tmp/dracut.qFSaUy/initramfs/sbin/
/var/tmp/dracut.qFSaUy/initramfs/sbin/
/var/tmp/dracut.qFSaUy/initramfs/sbin/nologin
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.minix
/var/tmp/dracut.qFSaUy/initramfs/sbin/losetup
/var/tmp/dracut.qFSaUy/initramfs/sbin/mpathpersist
/var/tmp/dracut.qFSaUy/initramfs/sbin/rdsosreport
/var/tmp/dracut.qFSaUy/initramfs/sbin/lsmod
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.ext2
/var/tmp/dracut.qFSaUy/initramfs/sbin/blkid
/var/tmp/dracut.qFSaUy/initramfs/sbin/modprobe
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.zfs
/var/tmp/dracut.qFSaUy/initramfs/sbin/dmsetup
/var/tmp/dracut.qFSaUy/initramfs/sbin/tracekomem
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck
/var/tmp/dracut.qFSaUy/initramfs/sbin/insmod
/var/tmp/dracut.qFSaUy/initramfs/sbin/zfs
/var/tmp/dracut.qFSaUy/initramfs/sbin/swapoff
/var/tmp/dracut.qFSaUy/initramfs/sbin/multipathd
/var/tmp/dracut.qFSaUy/initramfs/sbin/rmmod
/var/tmp/dracut.qFSaUy/initramfs/sbin/reboot
/var/tmp/dracut.qFSaUy/initramfs/sbin/partx
/var/tmp/dracut.qFSaUy/initramfs/sbin/e2fsck
/var/tmp/dracut.qFSaUy/initramfs/sbin/zpool
/var/tmp/dracut.qFSaUy/initramfs/sbin/kpartx
/var/tmp/dracut.qFSaUy/initramfs/sbin/poweroff
/var/tmp/dracut.qFSaUy/initramfs/sbin/initqueue
/var/tmp/dracut.qFSaUy/initramfs/sbin/loginit
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.ext3
/var/tmp/dracut.qFSaUy/initramfs/sbin/depmod
/var/tmp/dracut.qFSaUy/initramfs/sbin/init
/var/tmp/dracut.qFSaUy/initramfs/sbin/dmeventd
/var/tmp/dracut.qFSaUy/initramfs/sbin/crypt-run-generator
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.ext4
/var/tmp/dracut.qFSaUy/initramfs/sbin/fsck.cramfs
/var/tmp/dracut.qFSaUy/initramfs/sbin/modinfo
/var/tmp/dracut.qFSaUy/initramfs/sbin/halt
/var/tmp/dracut.qFSaUy/initramfs/sbin/mount.zfs
/var/tmp/dracut.qFSaUy/initramfs/sbin/insmodpost.sh
/var/tmp/dracut.qFSaUy/initramfs/sbin/multipath
```

and

```
ls -l /sbin/modprobe 
lrwxrwxrwx 1 root root 11 Dec 20 02:26 /sbin/modprobe -> ../bin/kmod
```

Reply:
`@sbindir@` is not required to be set to "/sbin", the default `./configure` prefix option is "/usr/local" so `@sbindir@`="/usr/local/sbin":

```
root@linux:/usr/src/zfs# git checkout dracut-encryption
Already on 'dracut-encryption'
Your branch is up-to-date with 'origin/dracut-encryption'.
root@linux:/usr/src/zfs# ( ./autogen.sh && ./configure --with-config=user && make -j`nproc` ) &> /dev/null 
root@linux:/usr/src/zfs# grep sbin etc/systemd/system/zfs-import-cache.service
ExecStartPre=/usr/local/sbin/modprobe zfs
ExecStart=/usr/local/sbin/zpool import -c /usr/local/etc/zfs/zpool.cache -aN
ExecStartPost=/usr/local/bin/bash -c "/usr/bin/systemctl set-environment BOOTFS=$(/usr/local/sbin/zpool list -H -o bootfs)"
root@linux:/usr/src/zfs# 
```

While it's true a properly packaged ZFS will indeed provide `/usr/local/sbin/zpool` the `modprobe` binary will not be found by systemd in "/usr/local/sbin". Same goes for `@bindir@`/`bash`.

Also please add a `Signed-off-by` to the commit message to make the STYLE builder happy.

Reply:
ah, guess I do have to hard code paths then, ok